### PR TITLE
[migration phase 1] CSIMaxVolumeLimitChecker as filter plugin

### LINKED ISSUE #83637: [migration phase 1] CSIMaxVolumeLimitChecker as filter plugin
**What would you like to be added**:

Add a filter plugin that calls into CSIMaxVolumeLimitChecker predicate, example #83460

**Why is this needed**:

Part of  #83554

/sig scheduling
/help
/priority important-soon

### PR DESCRIPTION
**Pull request by notpad**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #83637

*Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by notpad** [comment]
Ready for review

**Comment by draveness** [comment]
Can we hold this PR for https://github.com/kubernetes/kubernetes/pull/82072 to get merged, I can't remember how many times I rebased the master... Thanks

**Comment by notpad** [comment]
sure :)

File: pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// Name is the name of the plugin used in the plugin registry and configurations.
+const Name = "NodeVolumeLimits"
+
+// Name returns name of the plugin. It is used in logs, etc.
+func (pl *NodeVolumeLimits) Name() string {
+	return Name
+}
+
+// Filter invoked at the filter extension point.
+func (pl *NodeVolumeLimits) Filter(_ *framework.CycleState, pod *v1.Pod, nodeInfo *nodeinfo.NodeInfo) *framework.Status {
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits.go]: add `ctx context.Context` as the first argument

File: pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+			} else {
+				defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigration, false)()
+				defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.CSIMigrationAWS, false)()
+			}
+
+			p := New(getFakeCSINodeInfo(csiNode),
+				getFakeCSIPVInfo(test.filterName, test.driverNames...),
+				getFakeCSIPVCInfo(test.filterName, "csi-sc", test.driverNames...),
+				getFakeCSIStorageClassInfo("csi-sc", test.driverNames[0]))
+			gotStatus := p.(framework.FilterPlugin).Filter(nil, test.newPod, node)
```
Comment (draveness) [inline pkg/scheduler/framework/plugins/nodevolumelimits/node_volume_limits_test.go]: context.Background()

**Review by draveness — COMMENTED** [review]
Thanks, the PR needs to add context.Context as the first argument in Filter method.

**Comment by draveness** [comment]
I'll label the lgtm after the rebase, thanks!

**Comment by draveness** [comment]
Please rebase the master

**Comment by notpad** [comment]
Code rebased.  /cc @draveness @ahg-g

**Pull request merged**